From de7850ea70b1d71d73c4649a636a0a4f0058e88e Mon Sep 17 00:00:00 2001
From: houxiaoni <houxiaoni@allwinnertech.com>
Date: Tue, 14 Jul 2020 11:22:16 +0800
Subject: [PATCH 11/12] Add -lvchostif to link when using -lEGL on rpi

This is required because libEGL from userland uses sybols
from this library.

lib/libEGL.so.1.0.0                                                                                                                                                                                                                              121: 00000000     0 FUNC    GLOBAL DEFAULT  UND vc_dispmanx_element_add
  1552: 00000000     0 FUNC    GLOBAL DEFAULT  UND vc_dispmanx_element_add

These symbols are provided by libvchostif as seen below

lib/libvchostif.so
   252: 0000b161   192 FUNC    GLOBAL DEFAULT    9 vc_dispmanx_element_add
   809: 0000b161   192 FUNC    GLOBAL DEFAULT    9 vc_dispmanx_element_add

With this explicit link, plugins fail during runtime

(gst-plugin-scanner:571): GStreamer-WARNING **: Failed to load plugin '/usr/lib/gstreamer-1.0/libgstomx.so': Error relocating /usr/lib/libgstgl-1.0.so.0: vc_dispmanx_element_add: symbol not found
(gst-plugin-scanner:571): GStreamer-WARNING **: Failed to load plugin '/usr/lib/gstreamer-1.0/libgstopengl.so': Error relocating /usr/lib/libgstgl-1.0.so.0: vc_dispmanx_element_add: symbol not found

Upstream-Status: Pending
Change-Id: I179f37993ab12a1095912500cc5fd826dc6828a4
Signed-off-by: houxiaoni <houxiaoni@allwinnertech.com>
---
 m4/gst-gl.m4 | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/m4/gst-gl.m4 b/m4/gst-gl.m4
index c9c7602..dd3e15b 100644
--- a/m4/gst-gl.m4
+++ b/m4/gst-gl.m4
@@ -231,7 +231,7 @@ case $host in
                             HAVE_EGL=yes
                             HAVE_GLES2=yes
                             HAVE_EGL_RPI=yes
-                            EGL_LIBS="-lbcm_host -lvcos -lvchiq_arm"
+                            EGL_LIBS="-lbcm_host -lvchostif -lvcos -lvchiq_arm"
                             EGL_CFLAGS=""
                             AC_DEFINE(USE_EGL_RPI, [1], [Use RPi platform])
                           ])
-- 
1.9.1

